### PR TITLE
Fixes #307 gazebo ros control

### LINKED ISSUE #307: Create a J0 transmission file based on transmission_interface::simple_transmission
For the hand we should try to use (for the moment to get Gazebo running):
- transmission_interface::simple_transmission for the normal joints
- a modification of transmission_interface::simple_transmission for J0

https://github.com/ros-controls/ros_control/blob/indigo-devel/transmission_interface/include/transmission_interface/simple_transmission.h

What is actually being instantiated when reading from the URDF is not transmission_interface::SimpleTransmission, but transmission_interface::SimpleTransmissionLoader.

https://github.com/ros-controls/ros_control/blob/indigo-devel/transmission_interface/include/transmission_interface/simple_transmission_loader.h

So maybe we will need to create a transmission_interface::J0TransmissionLoader as well.

### PR DESCRIPTION
**Pull request by AndriyPt**
@VahidAminZ checked functionality on the real hand and it seems to be working.

### CODE REVIEW

File: sr_description/hand/xacro/palm/palm.transmission.xacro [hunk 1]
```diff
@@ -15,16 +15,20 @@
...
+    </xacro:if>
     <!-- motor hand -->
     <xacro:unless value="${muscletrans}">
-      <transmission type="sr_mechanism_model/SimpleTransmission" name="${prefix}palm_transmission">
-      <actuator name="${prefix}WRJ1" />
-      <joint name="${prefix}WRJ1" />
-      <mechanicalReduction>1</mechanicalReduction>
+      <transmission name="${prefix}palm_transmission">
+        <type>sr_mechanism_model/SimpleTransmission</type>
+      <actuator name="${prefix}WRJ1">
```
Comment (toliver) [inline sr_description/hand/xacro/palm/palm.transmission.xacro]: Indent

File: sr_gazebo_sim/gazebo_sim_plugin.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<library path="lib/libsr_gazebo_sim">
+  <class name="sr_gazebo_sim/SrGazeboHWSim" type="sr_gazebo_sim::SrGazeboHWSim" base_class_type="gazebo_ros_control::RobotHWSim">
+    <description>This simulated hardware creates supports RobotState for ethercat based controllers.</description>
```
Comment (toliver) [inline sr_gazebo_sim/gazebo_sim_plugin.xml]: Review sentence (remove `creates`?).

**Comment by toliver** [comment]
I'm going to add a few fixes so that the simulation works with the UR arm.

**Comment by toliver** [comment]
I'm also going to format the cpp so that the indentation is 2 spaces (ROS style).

**Comment by toliver** [comment]
To test this in simulation, now we would need the latest version of https://github.com/ros-industrial/universal_robot.
Not the forked version.

**Comment by ugocupcic** [comment]
@toliver in that case edit the project.rosinstall to reflect this?

**Comment by toliver** [comment]
OK

**Comment by toliver** [comment]
Done

File: sr_gazebo_sim/src/gazebo_hardware_sim.cpp [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    }
+  }
+
+  registerInterface(&this->fake_state_);
+  ROS_INFO_STREAM("Registered fake state");
+}
+
+void SrGazeboHWSim::registerSecondHardwareInterface(std::vector<transmission_interface::TransmissionInfo> transmissions)
+{
+  for (unsigned int j = 0; j < transmissions.size(); j++)
```
Comment (VahidAminZ) [inline sr_gazebo_sim/src/gazebo_hardware_sim.cpp]: For loops are size_t, unsigned int and unsigned. Can we stick to one or is it intentional?

File: sr_gazebo_sim/src/gazebo_hardware_sim.cpp [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+ *
+ */
+
+#include "sr_gazebo_sim/gazebo_hardware_sim.h"
+
+#include <string>
+#include <vector>
+#include <map>
+
+using boost::ptr_unordered_map;
```
Comment (VahidAminZ) [inline sr_gazebo_sim/src/gazebo_hardware_sim.cpp]: is it used anywhere?

File: sr_gazebo_sim/src/gazebo_hardware_sim.cpp (hunk 3 above)
Comment (toliver) [inline sr_gazebo_sim/src/gazebo_hardware_sim.cpp]: Fixed this one.

File: sr_gazebo_sim/src/gazebo_hardware_sim.cpp [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+  }
+
+  return result;
+}
+
+void SrGazeboHWSim::readSim(ros::Time time, ros::Duration period)
+{
+  gazebo_ros_control::DefaultRobotHWSim::readSim(time, period);
+
+  for (unsigned int j = 0; j < n_dof_; ++j)
```
Comment (toliver) [inline sr_gazebo_sim/src/gazebo_hardware_sim.cpp:184]: Changed the unsigned to unsigned int, as `n_dof_` is an unsigned int.
I think unsigned is the same as unsigned int, but maybe this way it's more clear?

**Pull request merged**